app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -40,6 +54,18 @@ def smart_proxy_ids
         scoped_search :relation => :host_traces, :on => :helper, :complete_value => true, :rename => :trace_helper, :only_explicit => true
       end
 
+      def correct_kickstart_repository
```

Comment:
I'm not a fan of two identical methods. Surely we have shared host/hostgroup stuff somewhere to DRY this up?

Reply:
We don't have any shared concerns today, I want this to be an easy cherry-pick, so I'd rather not do any big refactoring here.